src/BeamAdapter/gpu/cuda/CudaInstanciations.cpp

```diff
@@ -0,0 +1,97 @@
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <SofaCUDA/sofa/gpu/cuda/CudaTypes.h>
+
+#include <BeamAdapter/component/WireBeamInterpolation.inl>
+#include <BeamAdapter/component/BeamInterpolation.inl>
+#include <BeamAdapter/component/engine/WireRestShape.inl>
+#include <BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl>
+#include <BeamAdapter/component/controller/InterventionalRadiologyController.inl>
+#include <BeamAdapter/component/mapping/AdaptiveBeamMapping.inl>
+#include <BeamAdapter/component/mapping/MultiAdaptiveBeamMapping.inl>
+
+#include <sofa/core/behavior/Mass.inl>
+#include <sofa/core/Mapping.inl>
+#include <sofa/component/controller/MechanicalStateController.inl>
+#include <sofa/component/constraint/projective/FixedConstraint.inl> // for InterventionalRadiologyController
+
+#include <sofa/core/ObjectFactory.h>
+
+using namespace sofa::gpu::cuda;
+
+namespace sofa::component::fem::_beaminterpolation_
+{
+    template class SOFA_BEAMADAPTER_API BeamInterpolation<CudaRigid3dTypes>;
+} // namespace sofa::component::fem::_beaminterpolation_
+
+namespace sofa::component::fem::_wirebeaminterpolation_
+{
+    template class SOFA_BEAMADAPTER_API WireBeamInterpolation<CudaRigid3dTypes>;
+} // namespace sofa::component::fem::_beaminterpolation_
+
+namespace sofa::component::engine::_wirerestshape_
+{
+    template class SOFA_BEAMADAPTER_API WireRestShape<CudaRigid3dTypes>;
+} // namespace sofa::component::engine::_wirerestshape_
+
+namespace sofa::component::controller::_interventionalradiologycontroller_
+{
+    template class SOFA_BEAMADAPTER_API InterventionalRadiologyController<CudaRigid3dTypes>;
+} // namespace sofa::component::controller::_interventionalradiologycontroller_
+
+namespace sofa::component::mapping::_adaptivebeammapping_
+{
+    template class SOFA_BEAMADAPTER_API AdaptiveBeamMapping<CudaRigid3dTypes, defaulttype::Vec3Types>;
+} // namespace sofa::component::mapping::_adaptivebeammapping_
+
+namespace sofa::component::mapping
+{
+    template class SOFA_BEAMADAPTER_API MultiAdaptiveBeamMapping<CudaRigid3dTypes, defaulttype::Vec3Types>;
+} // namespace sofa::component::mapping
+
+
+namespace sofa::gpu::cuda
+{
+
+int CudaBeamInterpolationClass = core::RegisterObject("Adaptive Beam Interpolation")
```

Comment:
The description you give will be appended to the already existing description (I think). In SofaCUDA, the description of all components is "Supports GPU-side computations using CUDA", so that the final description is CPUDescription + "Supports GPU-side computations using CUDA"